public/javascripts/youtubeLink.js

```diff
@@ -0,0 +1,38 @@
+/*
+  youtube link always has 'v={code}' in the url where a unique code refers to a video.
+  in order to be able to embed youtube video in html we need to include iframe element with
+  src='https://www.youtube.com/embed/{code}'.
+  therefore we need to to extract only the code from youtube url.
+  there are two cases of youtube link:
+      1. https://www.youtube.com/watch?v=O0VN0pGgBZM&t=180s
+         in this case we need to get the substring until the index of '&'
+      2. https://www.youtube.com/watch?v=O0VN0pGgBZM
+         in this case we need to get the subtring until the end of the string
+  this function will take original youtube link and return the embed link that we need to put in html
+*/
+function encoder(youtubeLink) {
+    if (youtubeLink === '') {
+        return youtubeLink
+    } else {
+        var startIndex = youtubeLink.indexOf('v=') + 2;
+        var endIndex = youtubeLink.indexOf('&');
+        var embedLink = 'https://www.youtube.com/embed/';
+        if (endIndex !== -1) {
+            return embedLink + youtubeLink.substring(startIndex, endIndex);
+        } else {
+            return embedLink + youtubeLink.substring(startIndex);
+        }
+    }
+}
+
+
+//this function takes the embed link and return the original youtube link
+function decoder(embedLink) {
+    if (embedLink === '') {
+        return embedLink;
+    } else {
+        var startIndex = embedLink.indexOf('embed/') + 6;
```

Comment:
don't do +6. Put embed link as a string and use the length of the string. indexOf('embed/') + 'embed/'.length 
Except you can put embed/ in a variable.